internal/setup/openpgp_test.go

```diff
@@ -0,0 +1,269 @@
+package setup_test
+
+import (
+	"golang.org/x/crypto/openpgp/packet"
+	. "gopkg.in/check.v1"
+
+	"github.com/canonical/chisel/internal/setup"
+)
+
+type archiveKeyTest struct {
+	summary  string
+	armored  string
+	relerror string
+	pubKey   *packet.PublicKey
+}
+
+var archiveKeyTests = []archiveKeyTest{{
+	summary: "Armored data with one public key",
+	armored: testKey.ArmoredPublicKey,
+	pubKey:  testKey.PublicKey,
+}, {
+	summary:  "Armored data with two public keys",
+	armored:  twoPubKeysArmored,
+	relerror: ".*contains more than one public key.*",
+}, {
+	summary:  "Armored data with no public key",
+	armored:  armoredDataWithNoKeys,
+	relerror: ".*no public key.*",
+}, {
+	summary:  "Armored data with private key",
+	armored:  testKey.ArmoredPrivateKey,
+	relerror: ".*contains private key.*",
+}, {
+	summary: "Invalid armored data",
+	armored: `
+		Roses are red
+		Violets are blue
+	`,
+	relerror: ".*cannot decode armored data.*",
+}, {
+	summary:  "Empty armored data",
+	relerror: ".*cannot decode armored data.*",
+}}
+
+func (s *S) TestDecodeArchivePubKey(c *C) {
+	for _, test := range archiveKeyTests {
+		c.Logf("Summary: %s", test.summary)
+
+		pubKey, err := setup.DecodeArchivePublicKey([]byte(test.armored))
+		if test.relerror != "" {
+			c.Assert(err, ErrorMatches, test.relerror)
+			continue
+		} else {
+			c.Assert(err, IsNil)
+		}
```

Comment:
Can we remove the `else` here, it is not necessary and we are later going to execute the `assert` below.